### PR TITLE
appender: add test for append to a exists file

### PR DESCRIPTION
**Pull request by xiongjiwei**

### CODE REVIEW

File: appender.go [hunk 1]
```diff
@@ -130,6 +130,15 @@ func TestAppender(t *testing.T, store types.Storager) {
 			Convey("WriteAppend size should be equal to n", func() {
 				So(n, ShouldEqual, size)
 			})
+
+			content, _ = ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), size))
+			r = bytes.NewReader(content)
+			_, _ = ap.WriteAppend(o, r, size)
+			o, _ = store.Stat(path)
```
Comment (JinnyYi) [inline appender.go]: The object or file may not be retrieved  before `CommitAppend`, like `dropbox`. 
Maybe we can modify `When CommitAppend` case: `WriteAppend` twice and check the content and size(content length) after `Read`.

Comment (xiongjiwei) [reply appender.go]: sure

File: appender.go [hunk 2]
```diff
@@ -148,9 +148,15 @@ func TestAppender(t *testing.T, store types.Storager) {
 
 			size := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
 			content, _ := ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), size))
-			r := bytes.NewReader(content)
 
-			_, err = ap.WriteAppend(o, r, size)
+			r := bytes.NewReader(content[:size/2])
```
Comment (Xuanwo) [inline appender.go]: How about using the `content` twice instead of split them into halves? (which introduces more logic)

**Review by Xuanwo — APPROVED** [review]

File: appender.go [hunk 3]
```diff
@@ -162,13 +166,13 @@ func TestAppender(t *testing.T, store types.Storager) {
...
 			var buf bytes.Buffer
-			_, err = store.Read(path, &buf, pairs.WithSize(size))
+			_, err = store.Read(path, &buf, pairs.WithSize(size*2))
 
 			Convey("Read error should be nil", func() {
 				So(err, ShouldBeNil)
 			})
 			Convey("The content should be match", func() {
-				So(sha256.Sum256(buf.Bytes()), ShouldResemble, sha256.Sum256(content))
+				So(sha256.Sum256(buf.Bytes()), ShouldResemble, sha256.Sum256(append(content, content...)))
```
Comment (Xuanwo) [inline appender.go]: How about using `bytes.Repeat(content, 2)` here to make it more readable?

File: appender.go (hunk 2 above)
Comment (xiongjiwei) [reply appender.go]: sounds good.

File: appender.go (hunk 3 above)
Comment (xiongjiwei) [reply appender.go]: thanks

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**